examples/images_contours_and_fields/image_annotated_heatmap.py

```diff
@@ -104,20 +104,23 @@ def heatmap(data, row_labels, col_labels, ax=None,
     """
     Create a heatmap from a numpy array and two lists of labels.
 
-    Arguments:
-        data       : A 2D numpy array of shape (N,M)
-        row_labels : A list or array of length N with the labels
-                     for the rows
-        col_labels : A list or array of length M with the labels
-                     for the columns
-    Optional arguments:
-        ax         : A matplotlib.axes.Axes instance to which the heatmap
-                     is plotted. If not provided, use current axes or
-                     create a new one.
-        cbar_kw    : A dictionary with arguments to
-                     :meth:`matplotlib.Figure.colorbar`.
-        cbarlabel  : The label for the colorbar
-    All other arguments are directly passed on to the imshow call.
+    Parameters
+    ----------
+    data
+        A 2D numpy array of shape (N, M).
+    row_labels
+        A list or array of length N with the labels for the rows.
+    col_labels
+        A list or array of length M with the labels for the columns.
+    ax
+        A `matplotlib.axes.Axes` instance to which the heatmap is plotted.  If
+        not provided, use current axes or create a new one.  Optional.
+    cbar_kw
+        A dictionary with arguments to `matplotlib.Figure.colorbar`.  Optional.
```

Comment:
Applies to all parameters. Typically we would have the type and "optional" on the line of the paramter name, such as:
~~~
    cbar_kw : dict, optional
        Keyword arguments passed to `matplotlib.Figure.colorbar`.
~~~

Not sure how much I want to enforce this, but it's a deviation from the numpydoc style.

Reply:
I know, but once I decided not to include types, it felt strange just to put "optional" up there...

Reply:
Yeah, optional without type is not not good.